File: commonutils/src/main/java/net/maxsmr/commonutils/android/gui/fragments/alert/BaseProgressDialogFragment.java
package net.maxsmr.commonutils.android.gui.fragments.alert;

import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


public abstract class BaseProgressDialogFragment<L extends AlertDialogFragment.EventListener> extends AlertDialogFragment<L> {

    protected abstract int getProgressBarId();

    protected abstract int getLoadingMessageViewId();

    @Nullable
    protected ProgressBar progressBar;

    protected TextView loadingMessageView;

    @Override
    protected void onDialogCreated(@NotNull AlertDialog dialog) {
        super.onDialogCreated(dialog);
        if (customView != null) {
            final int progressBarId = getProgressBarId();
            if (progressBarId != 0) {
                progressBar = customView.findViewById(progressBarId);
            }
            final int loadingMessageId = getLoadingMessageViewId();
            if (loadingMessageId != 0) {
                loadingMessageView = customView.findViewById(loadingMessageId);
            }
        }
        if (loadingMessageView != null) {
            final String message = args.getString(Args.ARG_MESSAGE);
            loadingMessageView.setText(message);
        }
    }
}

File: commonutils/src/main/kotlin/net/maxsmr/commonutils/gui/fragments/dialogs/BaseProgressDialogFragment.kt
package net.maxsmr.commonutils.gui.fragments.dialogs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.ProgressBar
import android.widget.TextView
import androidx.annotation.IdRes

const val ARG_PROGRESS_BAR_ID = "BaseProgressTypedDialogFragment#ARG_PROGRESS_BAR_ID"
const val ARG_LOADING_MESSAGE_VIEW_ID = "BaseProgressTypedDialogFragment#ARG_LOADING_MESSAGE_VIEW_ID"

abstract class BaseProgressDialogFragment<D : Dialog> : TypedDialogFragment<D>() {

    protected var progressBar: ProgressBar? = null
        private set

    protected var loadingMessageView: TextView? = null
        private set

    override fun onDialogCreated(dialog: D) {
        super.onDialogCreated(dialog)
        customView?.let {
            val progressBarId: Int = args.getInt(ARG_PROGRESS_BAR_ID)
            if (progressBarId != 0) {
                progressBar = it.findViewById(progressBarId)
            }
            val loadingMessageId: Int = args.getInt(ARG_LOADING_MESSAGE_VIEW_ID)
            if (loadingMessageId != 0) {
                with(it.findViewById<TextView>(loadingMessageId)) {
                    if (this != null) {
                        val message = args.getString(ARG_MESSAGE)
                        text = message
                    }
                    loadingMessageView = this
                }
            }
        }
    }

    abstract class Builder<D : Dialog, F : BaseProgressDialogFragment<D>> @JvmOverloads constructor(
            @IdRes
            protected val progressBarId: Int = 0,
            @IdRes
            protected val loadingMessageViewId: Int = 0
    ) : TypedDialogFragment.Builder<D, F>() {

        override fun createArgs(context: Context): Bundle {
            return super.createArgs(context).apply {
                if (progressBarId != 0) {
                    putInt(ARG_PROGRESS_BAR_ID, progressBarId)
                }
                if (loadingMessageViewId != 0) {
                    putInt(ARG_LOADING_MESSAGE_VIEW_ID, loadingMessageViewId)
                }
            }
        }
    }
}